internal/consensus/state.go

```diff
@@ -1049,7 +1049,7 @@ func (cs *State) handleMsg(ctx context.Context, mi msgInfo, fsyncUponCompletion
 		// once proposal is set, we can receive block parts
 		err = cs.setProposal(msg.Proposal, mi.ReceiveTime)
 		// See if we can try creating the proposal block if keys exist
-		if err != nil && cs.config.GossipTransactionKeyOnly && cs.privValidatorPubKey != nil {
+		if err == nil && cs.config.GossipTransactionKeyOnly && cs.privValidatorPubKey != nil {
```

Comment:
error == nil means there is a valid proposal already
error != nil means we don't have a valid proposal

So this change means we will try build the blocks and complete for the proposal only if the we have a valid proposal, which make sense to me

Reply:
error == nil means we have a valid proposal, so we should try to build the block from mempool. otherwise we fall back to old behavior where we wait for blockparts to be gossiped